app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    our_list = []
+    for human in people:
+        person = Person(human["name"], human["age"])
+        if human.get("husband"):
+            person.husband = human["husband"]
```

Comment:
Instead of directly assigning `human["husband"]` to `person.husband`, you could optimize this by checking if the husband is already in `our_list` and directly assigning the `Person` object if it exists.